packages/grpc-js/src/resolver-dns.ts

```diff
@@ -298,10 +298,16 @@ class DnsResolver implements Resolver {
       IPV6_REGEX.exec(target) ||
       IPV6_BRACKET_REGEX.exec(target);
     if (ipMatch) {
+      if (ipMatch[2]) {
```

Comment:
Not sure if the resolver changes are related to the headers problem being fixed.

Reply:
The output of this function determines the value of the `:authority` header.